v2/api/storage/customizations/storage_account_extensions.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1alpha1api20210401storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/secrets"
+)
+
+var _ extensions.SecretsRetriever = &StorageAccountExtension{}
+
+func (ext *StorageAccountExtension) RetrieveSecrets(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger) ([]*v1.Secret, error) {
+
+	// This has to be the current storage version. It will need to be updated
+	// if the storage version changes.
+	typedObj, ok := obj.(*storage.StorageAccount)
```

Comment:
Add in a type assertion that `storage.StorageAccount` implements `Hub` - that way we'll get a compilation error here if/when the storage version changes.

Reply:
Did this. Great idea.